src/shogun/machine/gp/GaussianARDSparseKernel.h

```diff
@@ -71,7 +71,6 @@ class CGaussianARDSparseKernel: public CGaussianARDKernel
 private:
 	void initialize_sparse_kernel();
 
-#if defined(HAVE_EIGEN3) && defined(HAVE_LINALG_LIB)
```

Comment:
@yorkerlin Hey Wu, please don't do such checks in the future. They are really hard to automatically re-factor. Rather use two lines with `#ifdef`